src/client/auth/actions.js

```diff
@@ -33,14 +33,23 @@ function validateForm(fields) {
 }
 
 function validateCredentials(fields) {
-  // Simulate lengthy async action.
   return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      if (fields.password === 'pass1')
+
+    var xhr = new XMLHttpRequest();
+
+    xhr.open('POST', '/api/v1/auth/login', true);
+    xhr.setRequestHeader('Content-type', 'application/json');
+
+    // @TODO show how to handle different password/username server errors
+    xhr.onreadystatechange = () => {
```

Comment:
Superagent is no go, since it's very old and number of opened issues looks scary. I have no problem using isomorphic-fetch, post will be logged via dispatcher anyway, but I don't want to pollute package.json. So let's stick native solution, and I will add link to isomorphic-fetch into comment.

but it shouls not be